pkg/rpcservice/sample_create.go

```diff
@@ -0,0 +1,82 @@
+package rpcservice
+
+import (
+	"context"
+
+	"github.com/stripe/stripe-cli/pkg/samples"
+	"github.com/stripe/stripe-cli/rpc"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+var getSampleConfig = samples.GetSampleConfig
+var createSample = samples.Create
+
+// SampleCreate creates a sample at a given path with the selected integration, client language, and server language.
+func (srv *RPCService) SampleCreate(ctx context.Context, req *rpc.SampleCreateRequest) (*rpc.SampleCreateResponse, error) {
+	selectedConfig, err := getSelectedConfig(req)
+	if err != nil {
+		return nil, err
+	}
+
+	resultChan := make(chan samples.CreationResult)
+	go createSample(
+		srv.cfg.UserCfg,
+		req.SampleName,
+		selectedConfig,
+		req.Path,
+		req.ForceRefresh,
+		resultChan,
+	)
+
+	for res := range resultChan {
+		if res.Err != nil {
+			return nil, res.Err
+		}
+		if res.State == samples.Done {
+			return &rpc.SampleCreateResponse{
+				Path:        res.Path,
+				PostInstall: res.PostInstall,
+			}, nil
+		}
```

Comment:
could there be another state return or something that we care about or need to handle?

Reply:
Good question. Definitely the client wants to know about success and failure. I don't think the client should be concerned about other states because they are internal details, like files being copied and stuff like that.